ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/mapper/claimantresponse/ClaimantResponseMapper.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.hmcts.cmc.ccd.domain.mapper.claimantresponse;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.ccd.domain.CCDYesNoOption;
+import uk.gov.hmcts.cmc.ccd.domain.claimantresponse.CCDClaimantResponse;
+import uk.gov.hmcts.cmc.ccd.domain.claimantresponse.CCDClaimantResponseType;
+import uk.gov.hmcts.cmc.ccd.domain.claimantresponse.CCDFormaliseOption;
+import uk.gov.hmcts.cmc.ccd.domain.claimantresponse.CCDResponseAcceptation;
+import uk.gov.hmcts.cmc.ccd.domain.claimantresponse.CCDResponseRejection;
+import uk.gov.hmcts.cmc.ccd.domain.mapper.Mapper;
+import uk.gov.hmcts.cmc.ccd.domain.mapper.PaymentIntentionMapper;
+import uk.gov.hmcts.cmc.ccd.exception.MappingException;
+import uk.gov.hmcts.cmc.domain.models.claimantresponse.ClaimantResponse;
+import uk.gov.hmcts.cmc.domain.models.claimantresponse.ClaimantResponseType;
+import uk.gov.hmcts.cmc.domain.models.claimantresponse.FormaliseOption;
+import uk.gov.hmcts.cmc.domain.models.claimantresponse.ResponseAcceptation;
+import uk.gov.hmcts.cmc.domain.models.claimantresponse.ResponseRejection;
+
+@Component
+public class ClaimantResponseMapper implements Mapper<CCDClaimantResponse, ClaimantResponse> {
+
+    private final PaymentIntentionMapper paymentIntentionMapper;
+    private final CourtDeterminationMapper courtDeterminationMapper;
+
+    @Autowired
+    public ClaimantResponseMapper(PaymentIntentionMapper paymentIntentionMapper,
+                                  CourtDeterminationMapper courtDeterminationMapper) {
+
+        this.paymentIntentionMapper = paymentIntentionMapper;
+        this.courtDeterminationMapper = courtDeterminationMapper;
+    }
+
+    @Override
+    public CCDClaimantResponse to(ClaimantResponse claimantResponse) {
```

Comment:
You could also rely on Java overloading mechanism?

`public CCDClaimantResponse to(ResponseAcceptation claimantResponse)`
`public CCDClaimantResponse to(ResponseRejection claimantResponse)`

That way could save you from having to test the type and subsequently case the `claimantResponse`

Reply:
can't do that .. i will be breaking the contract with the interface

Reply:
another alternative is to have a spearate mapper for acceptance and rejection.. let me know if you prefer that.. i was trying to avoid it

Reply:
Ah I see! Leave it as it is then 👍